### PR TITLE
NER labeling of documents

### PR DESCRIPTION
**Pull request by ringgaard**
Task for labeling documents with silver-annotations using the NER annotators. The resolver now has a local mention context for adding local aliases to the context. I have added an abbreviation annotator for annotating entity names followed by an abbreviation for the name, e.g. "US Air Force (USAF)". The abbreviations are added to the local mention context together with first and last names for persons.

I have fixed a number of bug in the NER annotators including the remaining fixes from PR #342.

Bonus:
- Only open each panel once in document viewer
- Skip tables in Wikipedia document extraction
- Output from parse tool in LEX format

### CODE REVIEW

File: sling/nlp/ner/annotators.cc [hunk 1]
```diff
@@ -935,6 +980,125 @@ void DateAnnotator::Annotate(const PhraseTable &aliases, SpanChart *chart) {
   }
 }
 
+void AbbreviationAnnotator::Init() {
+  static const char *skipwords[] = {
+    "-", "&",
+    "of", "for", "and", "the", "in",
+    "ltd", "inc", "corp",
```
Comment (rahul1980) [inline sling/nlp/ner/annotators.cc]: also, "llc" and "gmbh"

File: sling/nlp/ner/annotators.cc [hunk 2]
```diff
@@ -198,13 +213,18 @@ void SpanTaxonomy::Init(Store *store) {
...
     {"Q21199",     SPAN_NATURAL_NUMBER},   // natural number
+    {"Q2360980",   SPAN_UNIT_OF_AMOUNT},   // unit of amount
     {"Q8142",      SPAN_CURRENCY},         // currency
     {"Q47574",     SPAN_UNIT},             // unit of measurement
     {"Q27084",     SPAN_UNIT},             // parts-per notation
     {"Q101352",    SPAN_FAMILY_NAME},      // family name
     {"Q202444",    SPAN_GIVEN_NAME},       // given name
     {"Q19838177",  SPAN_SUFFIX},           // suffix for person name
+    {"Q215627",    SPAN_PERSON},           // person
+    {"Q11032",     0},                     // newspaper
```
[the remark this answers is not in the document]
Comment (ringgaard) [reply sling/nlp/ner/annotators.cc]: It is to prevent newspapers to be labeled as ART.

File: sling/nlp/ner/annotators.cc (hunk 1 above)
Comment (ringgaard) [reply sling/nlp/ner/annotators.cc]: I have added these, but I think I will need to make all the word lists in the annotators language dependent so they can be configured in languages.sling. For example, for Danish we would like to add A/S, Aps, I/S, K/S, etc., but some of these may be too ambiguous for some languages.

File: sling/nlp/ner/annotators.cc [hunk 3]
```diff
@@ -996,11 +1163,31 @@ void SpanAnnotator::Annotate(const Document &document, Document *output) {
...
+                      << ": " << item.cost;
+          } else {
+            LOG(INFO) << chart.phrase(b, e) << ": " << item.cost;
+          }
+        }
+      }
+    }
+
+    // Extract best span covering to the output document.
     chart.Solve();
```
[the remark this answers is not in the document]
Comment (ringgaard) [reply sling/nlp/ner/annotators.cc]: Oops. Good catch!

File: sling/nlp/ner/resolver.h [hunk 4]
```diff
@@ -27,46 +27,71 @@
...
-struct ResolverNames : public SharedNames {
-  ResolverNames(Store *store) { CHECK(Bind(store)); }
+class ResolverContext;
 
-  Name n_popularity{*this, "/w/item/popularity"};
-  Name n_links{*this, "/w/item/links"};
+// Entity resolver.
+class EntityResolver {
+ public:
+  // Initialize entiry resolver.
```
[the remark this answers is not in the document]
Comment (ringgaard) [reply sling/nlp/ner/resolver.h]: fixed

[the remark this answers is not in the document]
Comment (ringgaard) [reply sling/nlp/ner/resolver.h]: same?

**Pull request merged**